### PR TITLE
Fix bugs with how dialogs are handled

### PR DESCRIPTION
**Pull request by csolbs24**
# Fix bugs with how dialogs are handled

## Description

This pr fixes two bugs with how dialogs are handled.

The first bug is that the confidence dialog used to not update as soon as it became visible. Instead you had to hover over any dialog to make it showup. This fixes the issue by updating the confidence dialog when it shows up, not when it is hovered.

The second is #148. The issue was introduced in #140 when the logic to get global edit candidates was updated. The previous change made it so that dialogs would only show up when the mouse was inside of an annotation's hitbox. Since point annotations' hitbox only consist of one point, it was impossible for the mouse position to be inside it. This was solved by checking if the mouse position is within a threshold of the point. The current threshold value is an arbitrarily chosen number that closely matches the previous behavior. 

## PR Checklist

- [x] Merged latest main
- [x] Version number in `package.json` has been bumped since last release
- [x] Version numbers match between package `package.json` and `src/version.js`
- [x] Ran `npm install` and `npm run build` AFTER bumping the version number
- [x] Updated documentation if necessary (currently just in `api_spec.md`)
- [x] Added changes to `changelog.md`

## Breaking API Changes

none

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -6301,6 +6287,26 @@ export class ULabel {
         }
     }
 
+    // Update the displayed annotation confidence
+    update_confidence_dialog() {
+        // Whenever the mouse makes the dialogs show up, update the displayed annotation confidence.
+        const current_subtask = this.subtasks[this.state["current_subtask"]]
+        const active_annotation_id = current_subtask["active_annotation"]
+        const active_annotation = current_subtask["annotations"]["access"][active_annotation_id]
+        const aacp = active_annotation["classification_payloads"] // aacp stands for active annotation classification payloads
```
Comment (joshua-dean) [inline src/index.js:6296]:
```suggestion
        /** The active annotation's classification payloads. */
        const aacp = active_annotation["classification_payloads"]
```
Two changes here: imperative phrasing, and JSDoc syntax (see [this](https://stackoverflow.com/questions/38708777/what-is-the-correct-jsdoc-syntax-for-a-local-variable)).
JSDoc will allow editors to render the description on hover like this:
image

This is useful for locals like this that are named as an abbreviation of something much more verbose. 
I wouldn't advocate mandatory JSDoc on all locals.

Comment (csolbs24) [reply src/index.js]: Yeah that's definitely a good call :+1:

**Review by joshua-dean — APPROVED** [review]

**Comment by TrevorBurgoyne** [comment]
The `point` dialog issue only occurs when a `point` loaded from the `resume_from` does not have a `line_size` argument provided AND the user doesn't provide a default `initial_line_size` argument in the ulabel config.

To handle this case, the better solution is to use the `get_line_size()` function in `process_resume_from()` instead of defaulting to the `ul.state["line_size"]` which could be `null`

**Comment by csolbs24** [comment]
Nice catch! That's definitely a much better solution.

**Review by TrevorBurgoyne — APPROVED** [review]
LGTM 👍

**Pull request merged**